internal/service/databaseuser/model_database_user.go

```diff
@@ -40,24 +40,24 @@ func NewMongoDBDatabaseUser(ctx context.Context, dbUserModel *TfDatabaseUserMode
 		OidcAuthType: dbUserModel.OIDCAuthType.ValueStringPointer(),
 		LdapAuthType: dbUserModel.LDAPAuthType.ValueStringPointer(),
 		DatabaseName: dbUserModel.AuthDatabaseName.ValueString(),
-		Roles:        NewMongoDBAtlasRoles(rolesModel),
-		Labels:       NewMongoDBAtlasLabels(labelsModel),
-		Scopes:       NewMongoDBAtlasScopes(scopesModel),
+		Roles:        conversion.NonEmptySliceToSlicePtr(NewMongoDBAtlasRoles(rolesModel)),
+		Labels:       conversion.NonEmptySliceToSlicePtr(NewMongoDBAtlasLabels(labelsModel)),
+		Scopes:       conversion.NonEmptySliceToSlicePtr(NewMongoDBAtlasScopes(scopesModel)),
```

Comment:
why not have the `New..` function return the pointer?

Reply:
I'm trying to minimize the changes here, then there will a PR per resource to fix the update bugs, and check the readonly / createonly attributes

Reply:
as Agusting said before we'll probably get rid of from all NonEmptySliceToSlicePtr in the next PRs